params/history_segment.go

```diff
@@ -0,0 +1,249 @@
+    }
+]`)
+)
+
+type HistoryBlock struct {
+	Number uint64
+	hash   common.Hash
+	TD     uint64
+}
+
+func NewHistoryBlock(num uint64, hash common.Hash, td uint64) HistoryBlock {
+	return HistoryBlock{
+		Number: num,
+		hash:   hash,
+		TD:     td,
+	}
+}
+
+type HistorySegment struct {
+	Index           uint64      `json:"index"`                    // segment index number
+	ReGenesisNumber uint64      `json:"re_genesis_number"`        // new history segment start at a finality block number, called ReGenesisNumber
```

Comment:
Maybe it would be a good idea not to confuse this with the genesis block, and rather call this `StartingBlockNumber` . And the field below `StartingBlockHash` .

Reply:
It's better to use re-genesis, the BEP emphasize that it a new finality range like genesis, so called "ReGenesis".

Reply:
What does ReGenesis mean ?

Reply:
It provides a finality block to start sync/store block from there, like genesis.